devolutions-gateway/src/streaming.rs

```diff
@@ -83,11 +86,22 @@ pub(crate) async fn stream_file(
             .await
             .with_context(|| format!("failed to open file: {path:?}"))?;
 
+        let path_extension = path.extension().unwrap();
```

Comment:
suggestion: `path.extension()` is called at many places. Maybe it would be worth moving this at the beginning of the function, with explicit handling of the `None` case.

```rust
let path_extension = path.extension().context("no extension found in the recording file path")?;
```

You’ll save a few `Some(…)` wrapping above as well, and remove an `unwrap()` to audit. (By the way, we have a warning emitted when `unwrap()` is used.)

Also consider mapping `path_extension` to a given `RecordingFileType`, and work with that for the rest of the function. Enums are more ergonomic to work with than strings.